crates/matrix-sdk-indexeddb/src/state_store/mod.rs

```diff
@@ -1211,27 +1147,52 @@ impl_state_store! {
     }
 
     async fn get_user_ids(&self, room_id: &RoomId) -> Result<Vec<OwnedUserId>> {
-        let ids: Vec<OwnedUserId> = self.get_stripped_user_ids_stream(room_id).await?;
+        let ids = self.get_user_ids_inner(room_id, RoomMemberships::empty(), true).await?;
         if !ids.is_empty() {
             return Ok(ids);
         }
-        self.get_user_ids_stream(room_id).await
+        self.get_user_ids_inner(room_id, RoomMemberships::empty(), false).await
     }
 
     async fn get_invited_user_ids(&self, room_id: &RoomId) -> Result<Vec<OwnedUserId>> {
         let ids: Vec<OwnedUserId> = self.get_stripped_invited_user_ids(room_id).await?;
         if !ids.is_empty() {
             return Ok(ids);
         }
-        self.get_invited_user_ids_inner(room_id).await
+        self.get_user_ids_inner(room_id, RoomMemberships::INVITE, false).await
     }
 
     async fn get_joined_user_ids(&self, room_id: &RoomId) -> Result<Vec<OwnedUserId>> {
         let ids: Vec<OwnedUserId> = self.get_stripped_joined_user_ids(room_id).await?;
         if !ids.is_empty() {
             return Ok(ids);
         }
-        self.get_joined_user_ids_inner(room_id).await
+        self.get_user_ids_inner(room_id, RoomMemberships::JOIN, false).await
+    }
+}
+
+/// A room member.
+#[derive(Debug, Serialize, Deserialize)]
+struct RoomMember {
+    user_id: OwnedUserId,
+    membership: u16,
```

Comment:
The `bitflags` crate seems to have a `serde` feature flag, why not use that?

Reply:
Because it stores the flags as a human-readable string if we serialize to a human-readable format (which is the case with JSON), this allows to store it in a more compact format.